src/main/scala/ch/epfl/bluebrain/nexus/kg/marshallers/instances.scala

```diff
@@ -74,30 +78,16 @@ package object instances extends FailFastCirceSupport {
     Marshaller.oneOf(marshallers: _*)
   }
 
-  private def statusCodeFrom(rejection: Rejection): StatusCode = rejection match {
-    case _: IsDeprecated             => StatusCodes.BadRequest
-    case _: ProjectIsDeprecated      => StatusCodes.BadRequest
-    case _: IncorrectTypes           => StatusCodes.BadRequest
-    case _: IllegalContextValue      => StatusCodes.BadRequest
-    case _: UnableToSelectResourceId => StatusCodes.BadRequest
-    case _: InvalidResource          => StatusCodes.BadRequest
-    case _: IncorrectId              => StatusCodes.BadRequest
-    case _: InvalidPayload           => StatusCodes.BadRequest
-    case _: InvalidJsonLD            => StatusCodes.BadRequest
-    case _: NotFileResource          => StatusCodes.BadRequest
-    case _: IllegalParameter         => StatusCodes.BadRequest
-    case _: MissingParameter         => StatusCodes.BadRequest
-    case _: Unexpected               => StatusCodes.InternalServerError
-    case _: UnexpectedState          => StatusCodes.InternalServerError
-    case _: ProjectNotFound          => StatusCodes.NotFound
-    case _: ProjectsNotFound         => StatusCodes.NotFound
-    case _: LabelsNotFound           => StatusCodes.NotFound
-    case _: NotFound                 => StatusCodes.NotFound
-    case _: OrganizationNotFound     => StatusCodes.NotFound
-    case InvalidResourceIri          => StatusCodes.NotFound
-    case _: IncorrectRev             => StatusCodes.Conflict
-    case _: AlreadyExists            => StatusCodes.Conflict
-    case _: DownstreamServiceError   => StatusCodes.BadGateway
-    case _: InvalidIdentity          => StatusCodes.Unauthorized
+  implicit class EitherTask[R <: Rejection, A](task: Task[Either[R, A]])(implicit s: Scheduler) {
+    def runWithStatus(code: StatusCode): Future[Either[R, (StatusCode, A)]] =
+      task.map(_.map(code -> _)).runToFuture
+  }
+
+  implicit class OptionTask[A](task: Task[Option[A]])(implicit s: Scheduler) {
+    def runNotFound: Future[A] =
+      task.flatMap {
+        case Some(a) => Task.pure(a)
+        case None    => Task.raiseError(KgError.NotFound())
```

Comment:
NotFound contains an optional ref just because of this call.
Maybe we could have a typeclass
```trait Referenced[A]{
   def apply(value: A): Ref 
}
```
And then have the signature here as `OptionTask[A: Referenced]`